### PR TITLE
Support controller manager framework

### PR DESCRIPTION
**Pull request by AmitRoushan**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
> /kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
> /kind enhancement

/kind enhancement

**What this PR does / why we need it**:
The PR contribute to support generic controller manager framework for kahu controller manager
**Which issue(s) this PR fixes**:

Fixes #

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
> /kind NOT-TESTED
/kind TESTED
**Test Report**:

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by AmitRoushan** [comment]
@PravinRanjan10 @joseph-v @sushanthakumar 
PTAL

**Review by sushanthakumar — DISMISSED** [review]
LGTM

**Review by PravinRanjan10 — CHANGES_REQUESTED** [review]

**Review by sushanthakumar — APPROVED** [review]
LGTM

File: controllers/controller.go [hunk 1]
```diff
@@ -90,8 +92,15 @@ func (ctrl *controller) SetInformerSyncWaiters(waiters ...cache.InformerSynced)
 	return ctrl
 }
 
-func (ctrl *controller) Build() Controller {
-	return ctrl
+func (ctrl *controller) Build() (Controller, error) {
```
[the remark this answers is not in the document]
Comment (AmitRoushan) [reply controllers/controller.go:95]: The ```Build``` is part of builder pattern to get ```Controller``` interface. IMO, the method is valid and required to generate controller.
Note: Corrected duplicate ```handle``` and ```resyncHandler``` error check

**Review by PravinRanjan10 — APPROVED** [review]
LGTM

**Pull request merged**